Source/Core/Core/PowerPC/GDBThread.cpp

```diff
@@ -0,0 +1,1274 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// Originally written by Sven Peter <sven@fail0verflow.com> for anergistic.
+
+#include "Core/PowerPC/GDBThread.h"
+
+#include "Common/CommonTypes.h"
+#include "Common/Event.h"
+#include "Common/Logging/Log.h"
+#include "Common/Thread.h"
+#include "Core/BootManager.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "Core/HW/CPU.h"
+#include "Core/HW/Memmap.h"
+#include "Core/Host.h"
+#include "Core/PowerPC/Interpreter/Interpreter_FPUtils.h"
+#include "Core/PowerPC/JitInterface.h"
+#include "Core/PowerPC/PowerPC.h"
+
+#include <fcntl.h>
+#include <signal.h>
+#include <stdio.h>
+#include <string.h>
+#ifdef _WIN32
+#include <iphlpapi.h>
+#include <ws2tcpip.h>
+#else
+#include <netinet/in.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+#define SD_BOTH SHUT_RDWR
+#endif
+#include <stdarg.h>
+
+#define fail(msg)                                                                                  \
+  {                                                                                                \
+    ERROR_LOG(GDB_THREAD, msg);                                                                    \
+    GDBDeinit();                                                                                   \
+    return;                                                                                        \
+  }
+
+#define failr(msg)                                                                                 \
+  {                                                                                                \
+    ERROR_LOG(GDB_THREAD, msg);                                                                    \
+    GDBDeinit();                                                                                   \
+    return 0;                                                                                      \
+  }
+
+#define GDB_MAX_BP 100
+
+#define GDB_STUB_START '$'
+#define GDB_STUB_END '#'
+#define GDB_STUB_ACK '+'
+#define GDB_STUB_NAK '-'
+
+#ifdef _WIN32
+#ifndef MSG_WAITALL
+#define MSG_WAITALL 8
+#endif
+#endif
+
+#define REGISTER_ID(category, index) ((category << 8) | index)
```

Comment:
I don't see this one being used?

Reply:
Inherited from plugin code where client implementation resides.  Will remove.